schemas/2.3.0.json

```diff
@@ -0,0 +1,1498 @@
+        }
+      }
+    },
+    "ReferenceObject": {
+      "type": "string",
+      "format": "uri-reference"
+    },
+    "info": {
+      "type": "object",
+      "description": "General information about the API.",
+      "required": [
+        "version",
+        "title"
+      ],
+      "additionalProperties": false,
+      "patternProperties": {
+        "^x-[\\w\\d\\.\\-\\_]+$": {
+          "$ref": "#/definitions/specificationExtension"
+        }
+      },
+      "properties": {
+        "title": {
+          "type": "string",
+          "description": "A unique and precise title of the API."
+        },
+        "version": {
+          "type": "string",
+          "description": "A semantic version number of the API."
+        },
+        "description": {
+          "type": "string",
+          "description": "A longer description of the API. Should be different from the title. CommonMark is allowed."
+        },
+        "termsOfService": {
+          "type": "string",
+          "description": "A URL to the Terms of Service for the API. MUST be in the format of a URL.",
+          "format": "uri"
+        },
+        "contact": {
+          "$ref": "#/definitions/contact"
+        },
+        "license": {
+          "$ref": "#/definitions/license"
+        }
+      }
+    },
+    "contact": {
+      "type": "object",
+      "description": "Contact information for the owners of the API.",
+      "additionalProperties": false,
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The identifying name of the contact person/organization."
+        },
+        "url": {
+          "type": "string",
+          "description": "The URL pointing to the contact information.",
+          "format": "uri"
+        },
+        "email": {
+          "type": "string",
+          "description": "The email address of the contact person/organization.",
+          "format": "email"
+        }
+      },
+      "patternProperties": {
+        "^x-[\\w\\d\\.\\-\\_]+$": {
+          "$ref": "#/definitions/specificationExtension"
+        }
+      }
+    },
+    "license": {
+      "type": "object",
+      "required": [
+        "name"
+      ],
+      "additionalProperties": false,
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the license type. It's encouraged to use an OSI compatible license."
+        },
+        "url": {
+          "type": "string",
+          "description": "The URL pointing to the license.",
+          "format": "uri"
+        }
+      },
+      "patternProperties": {
+        "^x-[\\w\\d\\.\\-\\_]+$": {
+          "$ref": "#/definitions/specificationExtension"
+        }
+      }
+    },
+    "server": {
+      "type": "object",
+      "description": "An object representing a Server.",
+      "oneOf" : [
+        { "required" : ["url", "protocol"] },
+        { "required" : ["$ref"] }
+      ],    
```

Comment:
This allows us to set required properties based on the presence of `$ref`. Either we require `url` and `protocol`, or only `$ref` to a server schema.

Reply:
But with that schema we can also write sucha a:

```
$ref: ...
variables:
  - ...
  - ...
```

I know that we wanna have such a functionality to override properties, but atm we don't clarify it in the spec, so we should support only single `$ref` and full Server Object. This same should be done with Channel object. 

EDIT: I also see problem that we allow to define the `$ref` inside `component/server`. I think that we should allow it only in the `#/servers/{server}` (also in the `#/components/channels`), but I am opened to suggestion.

Reply:
Regarding overriding properties, it is possible to do with the rest so (afaik) so I don't see why we should not allow it for the new ones (for the sake of consistency).

Also this PR is not aiming to introduce a breaking change so removing features, even though they are not named or specified in the spec, is not desired IMHO.

Reply:
Ok I saw rest of the occurrences of `ReferenceObject` and you're right 😂  Feature that we have but nobody know about that, I like that moment.

But also with current solution you cannot write:

```yaml
$ref: ... 
url: ...
protocol: ...
```

and I'm not sure if someone should have possibility to override url/protocol, or shouldn't 🤔

Reply:
> But also with current solution you cannot write:

You mean with `2.2.0` right? Indeed you can't. That's exactly what this PR introduces.
And I think we should give total freedom on what the users wants to do, if they want to override the protocol, URL, etc its on their own (yes, we don't officially support that in our spec)

Reply:
No, I mean your PR. `oneOf` works in this way that only one item in the array must be valid, so if you pass:

```yaml
$ref: ... 
url: ...
protocol: ...
```

you exactly follow by two schemas in `oneOf`, so your data/input is invalid.

You should change `oneOf` to the `anyOf` and it exactly means `at least one schema is valid`, not `only one schema is valid` :)

Reply:
That's a really good catch 👍  Fixed!